encoder/vaapiencoder_h264.cpp

```diff
@@ -743,13 +743,21 @@ void VaapiEncoderH264::resetParams ()
     DEBUG("resetParams, ensureCodedBufferSize");
     ensureCodedBufferSize();
 
+    if (intraPeriod() == 0) {
+        ERROR("intra period must larger than 0");
+        assert(0);
```

Comment:
Better return error here. assert should use for bug only. It's not user friendly if we assert on input value on core library